worker/src/RTC/Producer.cpp

```diff
@@ -73,12 +73,27 @@ namespace RTC
 	        {
 	            MS_DEBUG_TAG(rtp, "XXXXX creating producer bin log. lively=%s", lively.ToStr().c_str());
 
-	            this->binLog.InitLog('p', lively.callId, lively.id);
+                std::string const callId = lively.callId;
+                std::string const producerId = lively.id;
+
+                std::string userId; //default ""
+                if (data.contains("appData")) {
+                    json const& rAppData = data["appData"];
+                    userId = Lively::GetUserIdFromAppData(rAppData);
+                }
+                if (userId.empty()) {
+                    MS_WARN_TAG(rtp, "producer create missing appdata or user info, defaulting to 0 for userId");
+                    userId = "0";
+                }
+
+                this->binLog.InitLog([callId, producerId, userId](uint64_t timestamp) -> std::string {
```

Comment:
why passing a function then storing the function instead of simply passing a string template ?